// Proto file representing the EDB Path class

syntax = "proto3";

package ansys.api.edb.v1;

import "edb_messages.proto";
import "point_data.proto";
import "refs.proto";

enum PathEndCapStyle {
  ROUND = 0;
  FLAT = 1;
  EXTENDED = 2;
  CLIPPED = 3;
  INVALID_END_CAP = -1;
}

enum PathCornerStyle {
  ROUND_CORNER = 0;
  SHARP_CORNER = 1;
  MITER_CORNER = 2;
}

service PathService {
  rpc Create (PathCreationMessage) returns (EDBObjMessage) {}
}

// Message for Create
message PathCreationMessage {
  EDBObjMessage layout = 1;
  LayerRefMessage layer = 2;
  NetRefMessage net = 3;
  ValueMessage width = 4;
  PathEndCapStyle end_cap1 = 5;
  PathEndCapStyle end_cap2 = 6;
  PathCornerStyle corner = 7;
  PointsMessage points = 8;
}